index.js

```diff
@@ -106,20 +138,50 @@ function testGenResponse(swagger, path, operation, response, config) {
  * @param  {string} path API path to generate tests for
  * @param  {string} operation operation of the path to generate tests for
  * @param  {json} config configuration for testGen
+ * @param  {json} parameters json for different parameters
  * @returns {string|Array} set of all tests for a path's operation
  */
-function testGenOperation(swagger, path, operation, config) {
+function testGenOperation(swagger, path, operation, config, parameters) {
   var responses = swagger.paths[path][operation].responses;
   var result = [];
   var res;
+  var param;
+  var parameterType = swagger.paths[path][operation].parameters;
+  var type;
 
+  // body parameters to payload
+  if (swagger.paths[path][operation].hasOwnProperty('parameters')) {
+    // process different parameters
+    for (param in parameterType) {
+      if (parameterType.hasOwnProperty(param)) {
+        type = parameterType[param];
+        switch (type.in) {
+          case 'body':
+            parameters.bodyParameters.push(type);
+            break;
+          case 'query':
+            parameters.queryParameters.push(type);
+            break;
+          case 'form':
+            parameters.formParameters.push(type);
+            break;
+          case 'path':
+            parameters.bodyParameters.push(type);
```

Comment:
Initially I thought I need to handle the global parameters, without thinking the global one could only be path parameters. 
But now, i think since the global will be referenced, so I actually don't need to go over the whole yaml, but just get the parameters in the operation level, it will have all the extended parameters if the swagger is resolved, is that right Mohsen?